### ISSUE TITLE
Xamarin.iOS: System.ExecutionEngineException: Attempting to JIT compile method '(wrapper delegate-begin-invoke) System.IAsyncResult System.Func`2<Library.SomeDelegate`1<EEETestCase.AppDelegate>, bool>:begin_invoke_IAsyncResult__this___T_AsyncCallback_object (Library.SomeDelegate`1<EEETestCase.AppDelegate>,System.AsyncCallback,object)'

### ISSUE THREAD

**Issue by rolfbjarne** [mono/mono#14973]
## Steps to Reproduce

Run this project on an iOS device: [eeetestcase2.zip](https://github.com/mono/mono/files/3275149/eeetestcase2.zip)

This looks very similar to #14972, but the difference is that in this case the offending code has to be in a referenced assembly. If it's included in the executable itself, the code works just fine.

This comes from VSTS #[900808](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/900808).

### Current Behavior

An ExecutionEngineException, which is printed to the Application Output:

```
Got ExecutionEngineException when executing AsyncFire from referenced assembly: System.ExecutionEngineException: Attempting to JIT compile method '(wrapper delegate-begin-invoke) System.IAsyncResult System.Func`2<Library.SomeDelegate`1<EEETestCase.AppDelegate>, bool>:begin_invoke_IAsyncResult__this___T_AsyncCallback_object (Library.SomeDelegate`1<EEETestCase.AppDelegate>,System.AsyncCallback,object)' while running in aot-only mode. See https://docs.microsoft.com/xamarin/ios/internals/limitations for more information.

  at Library.DelegateHelper.AsyncFire[T] () [0x0000e] in <a300b5a252934c4d8462c65d592ee87e>:0 
  at EEETestCase.AppDelegate.FinishedLaunching (UIKit.UIApplication application, Foundation.NSDictionary launchOptions) [0x0002a] in /Users/rolf/Downloads/Archive/EEETestCase/Main.cs:35 

```

The app also turns red.

### Expected Behavior

No ExecutionEngineException (and green screen).

**Version Used**:

https://github.com/xamarin/xamarin-macios/commit/e89c3271f3c (which is mono/2019-02:  bab25c0a80f7a9ad8f1f18be227119384e3ec0a0).

I can also reproduce with the current Xamarin.iOS stable (v12.8.0.2 / d16-0).

**Comment by marek-safar** [comment]
@vargaz any ideas why external reference can make a difference here?

/cc @SamMonoRT

**Comment by vargaz** [comment]
Wasn't this fixed by the fix for #14972  ?

**Comment by rolfbjarne** [comment]
This is working in in xamarin-macios/master (a7cb29a4b72405e62100babcc14645a6b6fe207f)
